### PR TITLE
1.2 keyword references

### PR DESCRIPTION
**Pull request by ngilain**
[OO-FEATURE] Add keywords use tracking
https://github.com/open-orchestra/open-orchestra-mongo-libs/pull/50
https://github.com/open-orchestra/open-orchestra-model-interface/pull/225
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/643
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/233
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/282
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1862
https://github.com/open-orchestra/open-orchestra-bbcode-bundle/pull/25
https://github.com/open-orchestra/open-orchestra/pull/1027

### CODE REVIEW

File: Backoffice/Tests/Manager/KeywordToDocumentManagerTest.php [hunk 1]
```diff
@@ -111,13 +111,34 @@ class FakeKeyword implements KeywordInterface
...
 
-    public function setLabel($label) {
+    public function setLabel($label)
+    {
         $this->label = $label;
     }
+
+    public function addUseInEntity($entityId, $entityType)
+    {
+
```
Comment (alavieille) [inline Backoffice/Tests/Manager/KeywordToDocumentManagerTest.php]: remove this empty line

File: BackofficeBundle/DependencyInjection/Compiler/ReferenceCompilerPass.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Class ReferenceCompilerPass
+ */
+class ReferenceCompilerPass extends AbstractTaggedCompiler implements CompilerPassInterface
+{
+    /**
+     * @param ContainerBuilder $container
+     *
+     * @api
```
Comment (alavieille) [inline BackofficeBundle/DependencyInjection/Compiler/ReferenceCompilerPass.php]: @api ?

**Comment by ngilain** [comment]
@alavieille : update done

**Comment by alavieille** [comment]
:+1:

File: Backoffice/Reference/Strategies/AbstractKeywordReferenceStrategy.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+     */
+    protected function extractKeywordIdsFromConditionAttribute(array $attribute)
+    {
+        if (!isset($attribute['keywords'])) {
+            return array();
+        }
+
+        $string = $attribute['keywords'];
+        $parsedString = str_replace(' ', '', $string);
+        $parsedString = str_replace(array('(',')', 'AND', 'OR', 'NOT'), ' ', $parsedString);
```
Comment (itkg-nanne) [inline Backoffice/Reference/Strategies/AbstractKeywordReferenceStrategy.php]: AND, OR, NOT and brackets are defined in const (https://github.com/open-orchestra/open-orchestra-model-interface/blob/master/ModelInterface/Repository/RepositoryTrait/KeywordableTraitInterface.php#L15)
You can use same code as https://github.com/open-orchestra/open-orchestra-cms-bundle/blob/master/Backoffice/Form/DataTransformer/ConditionToReferenceKeywordTransformer.php#L28 to allow new adding operators.

**Comment by ngilain** [comment]
@itkg-nanne : update done

**Comment by itkg-nanne** [comment]
:+1:

**Pull request merged**